### ISSUE TITLE
Special rendering for intermittent water areas

### ISSUE THREAD

**Issue by matkoniecz** [gravitystorm/openstreetmap-carto#996]
@imagico proposed in #805 to render specially intermittent water areas.

intermittent tag is in the dabase, seasonal needs style file change.

quote from https://github.com/gravitystorm/openstreetmap-carto/issues/805#issuecomment-50901241

This should be extended to water areas, for example the current appearance of Australia is quite irritating:

http://www.openstreetmap.org/#map=6/-24.377/129.749

since nearly all of the water areas you can see there are intermittent.

Both intermittent=yes and seasonal=\* should be considered.

Current use according to taginfo:
- waterway=\* + intermittent=yes: 672000 times (most of this is NHD imports)
- waterway=\* + seasonal=*: 1200 times
- natural=water + intermittent=yes: 45000 times
- natural=water + seasonal=*: <1000 times

**Comment by Penegal** [comment]
The intermittent bodies of water are also often rendered with blue stripes and a dashed outline, at least in France. I don't know if it could be a better rendering, but it's a possibility.

**Comment by liotier** [comment]
Seasonality is a specific sort of intermittency - variation is according to seasons instead of more randomly weather-related. I came here to ask for waterways with seasonal=yes to be rendered the same way as intermittent=yes

For now, seasonal waterways appear as solid lines and it is a bit disconcerting.

Treating seasonal=yes the same way as intermittent=yes feels quite correct and it would also minimize the work necessary to take this extra tag into account.

This issue covers water bodies... Should I file a new issue to cover waterways ?

**Comment by dieterdreist** [comment]
sent from a phone

> Am 24.03.2016 um 13:19 schrieb Jean-Marc Liotier <notifications@github.com>:
> 
> Treating seasonal=yes the same way as intermittent=yes feels quite correct and it would also minimize the work necessary to take this extra tag into account.


I agree and generally support this, even if it might have to wait till we add hstore (if ever)

**Comment by kocio-pl** [comment]
> Both intermittent=yes and seasonal=* should be considered.

Because we already have hstore and are able to use non-standard tags too, we can do it quite easily now. I just wanted to ask if this is still the same (just adding seasonal=*, without "no" probably, the same as we use intermittent=yes) as I understand original proposition or something has changed in the meanwhile?

Currently seasonal has 108 828 uses together with waterway=* and 16 696 with natural=water, which makes it very popular tag. It's also growing fast in the last years:

![taghistory 7](https://user-images.githubusercontent.com/5439713/33253044-edd48c12-d341-11e7-8b25-14129871c500.png)

**Comment by Penegal** [comment]
Seems relevant to render both tags this way, as `seasonal` is merely a refinement of `intermittent`.

**Comment by Penegal** [comment]
Are there some things blocking this? Would be great to render intermittency.

**Comment by kocio-pl** [comment]
Probably the missing part is someone to make the code. Could you try?

**Comment by Penegal** [comment]
Oh, dear… I will try, as I have no real knowledge about CartoCSS, but, hey, I won't have any knowledge if I don't try.

**Comment by kocio-pl** [comment]
That's how I got into this whole thing. 😄 I still have only a basic understanding of CartoCSS, but dealing with SQL seems to be more important in osm-carto (even if I also don't know it too much). 

This case shouldn't be too hard, since you want to just copy intermittent tag behavior, not to create completely new code.

**Comment by Penegal** [comment]
> This case shouldn't be too hard

Well, we'll see… _(insert snapping noise of latex gloves here)_

**Comment by Penegal** [comment]
![image](https://user-images.githubusercontent.com/13777824/36216337-ca2301d0-11ae-11e8-9645-9cd1825450fb.png)

How about this way? It is an inner member of the meadow multipolygon, which should explain the grey lines.

**Comment by kocio-pl** [comment]
I guess very light blue background would work better for me. For me it's enough that stripes are wide.

**Comment by Penegal** [comment]
Do you mean a lighter blue between water-coloured stripes? I let it transparent to leave the eventual other landuse or natural area visible, as an intermittent water area is often used for something else when not flooded. That way, the stripes allow to see both usual landuse and flooded area.

**Comment by Penegal** [comment]
@kocio-pl : could you explain what you meant? I don't understand.

Other opinions about possible improvements are welcomed.

**Comment by kocio-pl** [comment]
Yes, I meant lighter blue instead of a land color. 

Could you give the examples when the area has a different use? I can't imagine such examples, so I tend to think my solution would be better. Reality check would help us to make decision.

**Comment by Penegal** [comment]
@kocio-pl : here is one:
![image](https://user-images.githubusercontent.com/13777824/36982649-a2cdbdfa-2090-11e8-9288-c28313ea9cc7.png)

Not as legible as I thought: the borders disappear way too notably between the stripes. Maybe with a dashed outline? I'll test that.

I don't reject your proposal, but, IMO, a plain color would mask another possible landuse, and I saw such areas frequently used for something else when not filled, so I prefer a way to display it if available.

**Comment by kocio-pl** [comment]
Great, thanks for testing!

**Comment by Penegal** [comment]
@kocio-pl : and with the dashed outline, the same as the one for intermittent streams (4,3):
![image](https://user-images.githubusercontent.com/13777824/36985920-e1b5d9d2-2098-11e8-829a-48ebe05df325.png)

Maybe with longer dashes (8,4):
![image](https://user-images.githubusercontent.com/13777824/36985801-81e83018-2098-11e8-977b-d28b4cadaefe.png)

**Comment by jragusa** [comment]
Why not use the Danger pattern with (or without) dashed stroke ?

Very large dash may increase focus on intermittent areas. On french map, dashes of intermittent area are very thin:
![intermittent](https://user-images.githubusercontent.com/9897203/36991179-340db25a-20a7-11e8-9854-97ec997cf850.png)

**Comment by dieterdreist** [comment]
2018-03-05 16:21 GMT+01:00 Penegal <notifications@github.com>:

> @kocio-pl <https://github.com/kocio-pl> : here is one:
> [image: image]
> <https://user-images.githubusercontent.com/13777824/36982649-a2cdbdfa-2090-11e8-9288-c28313ea9cc7.png>
>


I think the approach is OK, but the details should be adjusted. If you look
at the topographic map in this thread (5/3/2018, 19:02) that one works
nicely because the lines and distances are much finer (i.e. can show much
more detail). Also 0 degress hatching angles is more pleasing than the 60
we have here.

**Comment by kocio-pl** [comment]
> Very large dash may increase focus on intermittent areas. On french map, dashes of intermittent area are very thin:

Works for me.

**Comment by Penegal** [comment]
Then here it comes with horizontal 2px lines and dashed border:
![image](https://user-images.githubusercontent.com/13777824/37019233-d4b9c35e-2117-11e8-861c-002d0c23517d.png)

And without the dashed border:
![image](https://user-images.githubusercontent.com/13777824/37019302-19088e3c-2118-11e8-9dba-e50f6d870702.png)

The dashed border seems pretty useless with thin stripes, so I'll get rid of it. Is it OK, now? Anything to improve?

**Comment by kocio-pl** [comment]
For me that's very convincing and the border is not needed. Could you prepare a PR?

**Comment by matthijsmelissen** [comment]
Looks good to me too! Go ahead I’d say.

**Comment by kocio-pl** [comment]
I've learned that waiting too much does not help. We're discussing it for many days already and it will get more attention once the PR is ready.

**Comment by dieterdreist** [comment]
2018-03-06 11:26 GMT+01:00 Matthijs Melissen <notifications@github.com>:

> Looks good to me too! Go ahead I’d say.
>

+1, what might eventually be worth a consideration: how does this relate to
natural=wetland, which one is "wetter"?

**Comment by imagico** [comment]
For info: The horizontal hatching in French topographic maps indicates _Zone inondable_ - which means _area subject to flooding_ which is not the same as intermittent waterbodies in OSM - see http://www.viewranger.com/help/FranceLegend.pdf

Might be good to keep in mind regarding mapper feedback, in particular for French mappers.

**Comment by kocio-pl** [comment]
Good to know, however it's inevitable that we will use similar visual elements in the other meaning or context than some other maps. It might be more interesting if we have similar type of object so we could differentiate them - do we have "area subject to flooding" tagging scheme?

**Comment by dieterdreist** [comment]
2018-03-06 12:21 GMT+01:00 Christoph Hormann <notifications@github.com>:

> For info: The horizontal hatching in French topographic maps indicates *Zone
> inondable* - which means *area subject to flooding* which is not the same
> as intermittent waterbodies in OSM - see http://www.viewranger.com/
> help/FranceLegend.pdf
>


Is this about the difference between "generally not a waterbody, but
exceptionally flooded" and "generally a waterbody, but exceptionally dried
out"? Will this be verified statistically?

The wiki currently says about intermittent: "*a water
<https://wiki.openstreetmap.org/wiki/Key:water>=* body is not permanently
filled with water ... either regularly depending on seasons or based on
other conditions"*

Maybe this requires further specification?

**Comment by jragusa** [comment]
The _Zone inondable_ (area subject to flooding) feature in French map is not related to risk such as river flooding. It's restricted to body of water (pond, small lake) where the water level is susceptible to greatly fluctuate (e.g. flooding). In the example of the topographic map figured above, it's typically a seasonal small lake. It's dry during a long period because it's located in a karstic system. You won't find such hatching along rivers for example.

**Comment by imagico** [comment]
For better understanding -  _Zone inondable_ has a fairly specific meaning in French - see https://fr.wikipedia.org/wiki/Zone_inondable - that is strongly tied to French climate and the flooding patterns that are common in France.

A few examples of use of this in the classic IGN 25k map:

https://www.geoportail.gouv.fr/carte?c=-2.704535442894154,47.632134938389044&z=15&l0=ORTHOIMAGERY.ORTHOPHOTOS::GEOPORTAIL:OGC:WMTS(1)&l1=GEOGRAPHICALGRIDSYSTEMS.MAPS.SCAN-EXPRESS.STANDARD::GEOPORTAIL:OGC:WMTS(1)&l2=GEOGRAPHICALGRIDSYSTEMS.PLANIGN::GEOPORTAIL:OGC:WMTS(1)&l3=CADASTRALPARCELS.PARCELS::GEOPORTAIL:OGC:WMTS(1)&l4=GEOGRAPHICALGRIDSYSTEMS.MAPS::GEOPORTAIL:OGC:WMTS(1)&permalink=yes

https://www.geoportail.gouv.fr/carte?c=6.4734901652063925,43.83035184959559&z=16&l0=ORTHOIMAGERY.ORTHOPHOTOS::GEOPORTAIL:OGC:WMTS(1)&l1=GEOGRAPHICALGRIDSYSTEMS.MAPS.SCAN-EXPRESS.STANDARD::GEOPORTAIL:OGC:WMTS(1)&l2=GEOGRAPHICALGRIDSYSTEMS.PLANIGN::GEOPORTAIL:OGC:WMTS(1)&l3=CADASTRALPARCELS.PARCELS::GEOPORTAIL:OGC:WMTS(1)&l4=GEOGRAPHICALGRIDSYSTEMS.MAPS::GEOPORTAIL:OGC:WMTS(1)&permalink=yes

https://www.geoportail.gouv.fr/carte?c=4.451792947810736,44.3582615358346&z=15&l0=ORTHOIMAGERY.ORTHOPHOTOS::GEOPORTAIL:OGC:WMTS(1)&l1=GEOGRAPHICALGRIDSYSTEMS.MAPS.SCAN-EXPRESS.STANDARD::GEOPORTAIL:OGC:WMTS(1)&l2=GEOGRAPHICALGRIDSYSTEMS.PLANIGN::GEOPORTAIL:OGC:WMTS(1)&l3=CADASTRALPARCELS.PARCELS::GEOPORTAIL:OGC:WMTS(1)&l4=GEOGRAPHICALGRIDSYSTEMS.MAPS::GEOPORTAIL:OGC:WMTS(1)&permalink=yes

https://www.geoportail.gouv.fr/carte?c=3.528002352209444,43.31140581049013&z=15&l0=ORTHOIMAGERY.ORTHOPHOTOS::GEOPORTAIL:OGC:WMTS(1)&l1=GEOGRAPHICALGRIDSYSTEMS.MAPS.SCAN-EXPRESS.STANDARD::GEOPORTAIL:OGC:WMTS(1)&l2=GEOGRAPHICALGRIDSYSTEMS.PLANIGN::GEOPORTAIL:OGC:WMTS(1)&l3=CADASTRALPARCELS.PARCELS::GEOPORTAIL:OGC:WMTS(1)&l4=GEOGRAPHICALGRIDSYSTEMS.MAPS::GEOPORTAIL:OGC:WMTS(1)&permalink=yes

https://www.geoportail.gouv.fr/carte?c=-2.6182072228910243,47.521276412406195&z=15&l0=ORTHOIMAGERY.ORTHOPHOTOS::GEOPORTAIL:OGC:WMTS(1)&l1=GEOGRAPHICALGRIDSYSTEMS.MAPS.SCAN-EXPRESS.STANDARD::GEOPORTAIL:OGC:WMTS(1)&l2=GEOGRAPHICALGRIDSYSTEMS.PLANIGN::GEOPORTAIL:OGC:WMTS(1)&l3=CADASTRALPARCELS.PARCELS::GEOPORTAIL:OGC:WMTS(1)&l4=GEOGRAPHICALGRIDSYSTEMS.MAPS::GEOPORTAIL:OGC:WMTS(1)&permalink=yes

**Comment by Penegal** [comment]
Being French, I can tell you that what is mapped as _zone inondable_ is any area which is intermitently covered by water; that includes both floodable areas and intermittent water bodies. In fact, most areas mapped as _zone inondable_ are only intermittent water bodies, the rest being only the most flood-prone areas – not all of them, by far –, so you don't have to bother with the distinction between floodable areas and intermittent water bodies just because the french IGN confuses them on its maps.

**Comment by kocio-pl** [comment]
Another interesting thing to test: #1167.

**Comment by kocio-pl** [comment]
One more similar idea: #1547.
